arangod/IResearch/IResearchLinkMeta.cpp

```diff
@@ -391,30 +433,28 @@ bool IResearchLinkMeta::json(arangodb::velocypack::Builder& builder,
   }
 
   if (!mask || mask->_fields) {  // fields are not inherited from parent
+    arangodb::velocypack::Builder fieldBuilder;
     arangodb::velocypack::Builder fieldsBuilder;
+    Mask fieldMask(true); // output all non-matching fields
+    auto subDefaults = *this; // make modifable copy
 
-    {
-      arangodb::velocypack::ObjectBuilder fieldsBuilderWrapper(&fieldsBuilder);
-      arangodb::velocypack::Builder fieldBuilder;
-      Mask mask(true);  // output all non-matching fields
-      auto subDefaults = *this;
-
-      subDefaults._fields.clear();  // do not inherit fields and overrides
-                                    // overrides from this field
+    subDefaults._fields.clear(); // do not inherit fields and overrides from this field
+    fieldsBuilder.openObject();
 
       for (auto& entry : _fields) {
-        mask._fields = !entry.value()->_fields.empty();  // do not output empty fields on subobjects
+        fieldMask._fields = !entry.value()->_fields.empty(); // do not output empty fields on subobjects
+        fieldBuilder.openObject();
```

Comment:
IMHO using a second Builder instance would not be necessary here.
The entire data could also be built in a single Builder instance, even if it is nested.

Reply:
done